backend/src/database/program_classes.go

```diff
@@ -67,7 +68,37 @@ func (db *DB) UpdateProgramClass(ctx context.Context, content *models.ProgramCla
 	classLogEntries := models.GenerateChangeLogEntries(existing, content, "program_classes", existing.ID, content.UpdateUserID, ignoredFieldNames)
 	allChanges = append(allChanges, classLogEntries...)
 
+	statusChanged := (content.Status == models.Completed || content.Status == models.Cancelled) && existing.Status != content.Status
 	models.UpdateStruct(existing, content)
+	if statusChanged && len(existing.Events) > 0 {
+		var newest models.ProgramClassEvent
+		for i, ev := range existing.Events {
+			if i == 0 || ev.CreatedAt.After(newest.CreatedAt) { newest = ev }
+		}
+		loc, _ := time.LoadLocation(existing.Facility.Timezone)
+		cutoffDate := time.Now().In(loc)
+		until := cutoffDate.Format("20060102") + "T000000Z"
+		updatedRule := models.ReplaceOrAddUntilDate(newest.RecurrenceRule, until)
+		if updatedRule != newest.RecurrenceRule {
+			if err := trans.Model(&models.ProgramClassEvent{}).Where("id = ?", newest.ID).Update("recurrence_rule", updatedRule).Error; err != nil {
+				trans.Rollback()
+				return nil, newUpdateDBError(err, "program_class_events")
+			}
+			cle := models.NewChangeLogEntry("program_classes", "event_rescheduled_series", models.StringPtr(newest.RecurrenceRule), &updatedRule, existing.ID, content.UpdateUserID)
+			allChanges = append(allChanges, *cle)
```

Comment:
@JamesUnlocked - We don't need to record a changelog if we are updating the RRULE for the class (if cancelling/completing).